scripts/plot_dataframe.py

```diff
@@ -124,10 +124,12 @@ def analyse_and_plot_pairs(config: Dict[str, Any]):
         tickers = {}
         tickers[pair] = data
         dataframe = generate_dataframe(strategy, tickers, pair)
+        trades = None
+        if config["trade_source"] == "DB":
+            trades = load_trades_from_db(config["db_url"])
+        elif config["trade_source"] == "file":
+            trades = load_backtest_data(Path(config["exportfilename"]))
```

Comment:
I'm not sure this should be a switch with the type of the source. Is not it possible to use --exportfilename and --db-url if any is specified and make those options incompatible instead so that only one can be specified in the command line?

I don't like this new option, seems to be redundant... 👎

Reply:
Technically i agree with you - but there is no other way at the moment since db_url is ALWAYS in the configuration (and the same happens with exportfilename).

We need it as a default - so how do i determine which of the options was selected intentionally?
(it's also not really about defaults - see below)

It's also valid to specify `db_url` and `exportfilename` in the configuration (one is used during backtesting, the other during live/dry trading, so they don't collide apart from plotting).
Plotting however supports only one or the other (so it's not about cli arguments, really).

Honestly, this part was always broken (the working part switched back and forth, with each try to fix it).

We may remove this option in the future if we find a better solution - but it's not redundant, sorry.

Reply:
okay, let's leave it for now...